java/code/src/org/keyczar/KeyczarKey.java

```diff
@@ -84,6 +87,8 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Util.toInt(this.hash());
   }
+  
+  protected Stream cachedStream = null;
```

Comment:
You are right, this needs to be a StreamQueue, and every key refactored appropriately, that or remove caching all together